flashlight/nn/modules/Module.h

```diff
@@ -44,6 +44,17 @@ class Module {
    */
   std::vector<Variable> params_;
 
+  /**
+   * A golden reference of parameters (FP32) when mixed precision mode is
+   * enabled.
+   */
+  std::vector<af::array> paramsF32_;
+
+  /**
+   * Determines if the module is allowed to uses mixed precision.
+   */
+  bool useMixedPrecision_ = false;
+
```

Comment:
As above, we don't need to serialize the mixed precision flag and can always serialize the fp32 params for now. We can add functionality to change the precision of the weights we serialize later.